testing/types/doc.go

```diff
@@ -0,0 +1,4 @@
+/*
+Package types contains various interfaces for writing tests.
+*/
+package types
```

Comment:
As a newcomer to Go, I'm curious why this is pulled out into a separate file?

Reply:
I saw it being done in other packages in this repo, so I copied that pattern. I think the idea is to make an consistent place to look for the package docstring.

Reply:
It's a convention for package level documentation (https://go.dev/blog/godoc)

Reply:
In general types like this are put into their own package so that you don't end up with circular dependencies inside packages.

Reply:
More typical would be to have an internal type where it's used, like gotest.tools does:

https://github.com/gotestyourself/gotest.tools/blob/main/assert/assert.go#L77

I don't strongly have an issue with putting it in a common package, but it should be in an `/internal/types` subpackage.

Reply:
> As a newcomer to Go, I'm curious why this is pulled out into a separate file?

I think you only do this if you've got more than one file in the package. For a single-file package like this, I'd tend to do it inline.

Having it in a dedicated `doc.go` file makes it much easier to discover for those not using IDEs.

Reply:
https://github.com/circleci/ex/blob/main/testing/httprecorder/ginrecorder/middleware.go